### ISSUE TITLE
Cross Instance Persistent Themes and Settings

### ISSUE THREAD

**Issue by grravity** [SimonBrazell/privacy-redirect#141]
Requesting the ability to set themes on the redirect sites that support it, for example Bibliogram has different themes to choose from. My suggestion being to have all Bibliogram instances that a user visits, use that theme. Not sure entirely how this would be implemented (specifically detection of a Bibliogram instance).

Even better if settings can be applied cross instances as well.

I can understand if this is out of the scope of the project.

**Comment by SimonBrazell** [comment]
I tried to do this for invidious using the built in settings cookie but users didn't like it because of cookies... Is this like what you mean?

**Comment by grravity** [comment]
I believe so. I fail to understand their issue however, seeing it is a toggleable option

**Comment by SimonBrazell** [comment]
Most people had a problem with it because they were using cookie blockers, I suggested adding exceptions to the blockers for the Invidious sites they use but they wanted an option to redirect an invidious request to invidious instead, which seemed like a lot of effort for a trivial gain to me.

**Comment by grravity** [comment]
Ah, I thought you meant they had some privacy concern with it.
I'm not entirely sure. Have they suggested alternative ideas besides the redirect? That seems overly complicated.

**Comment by SimonBrazell** [comment]
I think people just don't trust cookies these days.

No I think that was the only suggestion. I'll see if the other providers have a similar option to Invidious with it's [preferences cookie](https://github.com/iv-org/documentation/blob/master/Preferences.md), which I can modify like I'm doing in the this [content script](https://github.com/SimonBrazell/privacy-redirect/blob/master/src/assets/javascripts/persist-invidious-prefs.js).

**Comment by tiotrom** [comment]
Maybe same could be used for SearX?

**Comment by grravity** [comment]
I agree with not trusting cookies, and I do block all cookies myself and unblock sparingly per site. However if this is the only/best way to achieve it, then it is still a toggleable option.

Also SearX would be nice too. However not sure because some instances have different settings and engines, that would need to be accounted for. I am tired of needing to reset SearX preferences everytime I want to swap SearX instances.

**Comment by microraptor** [comment]
Searx and Startpage allow to encode the settings in an URL parameter: [Startpage settings](https://www.startpage.com/do/settings)
Everytime the page is loaded with that parameter in the URL the preferences are applied automatically ([example](https://www.startpage.com/sp/search?prfe=dcd42af078991a56ea3eee07ce0585f1b76b10701b1de9d648a995b970aff6cd815298e162460541e0dc53d3a6165e00fd993a4349fc32a9cf9b58cf924b40f553ca2d2a2b5480341f2eb349&q=example)).
This is helpful, if cookies are deleted automatically. It would be great, if there is a field to define additional URL parameters in the Privacy Redirect settings.

**Comment by grravity** [comment]
They do mention it could be a privacy concern, so I think that should be considered
